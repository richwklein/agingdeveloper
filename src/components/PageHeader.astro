---
/**
 * @name PageHeader
 * Component used to render the page header.
 *
 * @param {Props} props - The properties of the page footer.
 * @return {Function} an Astro component factory function.
 *
 * @example
 * <PageHeader />
 */
import { Image } from 'astro:assets'
import type { CollectionEntry } from 'astro:content'

interface Props {
  site: CollectionEntry<'site'>
  class?: string
}

const { site, class: className } = Astro.props
---

<header
  class:list={[
    'sticky left-0 top-0 z-10',
    'bg-primary-dark px-4 py-4 text-primary-contrast shadow-lg',
    className,
  ]}
>
  <div class:list={['container mx-auto px-4 xl:max-w-screen-xl']}>
    <a href="/" class:list={['flex justify-start bg-transparent text-inherit']}>
      <div
        class:list={[
          'mb-2 mr-4 aspect-square h-12 min-h-12 w-12 min-w-12 sm:mb-0 md:h-20 md:w-20',
          'overflow-hidden rounded-lg ring-2 ring-secondary-light',
        ]}
      >
        <Image
          src={site.data.avatar}
          alt={site.data.title}
          loading={'eager'}
          fetchpriority={'high'}
          width={80}
          widths={[48, 80]}
          sizes={`(min-width: 768px) 80px, 48px`}
          class:list={['h-full w-full object-cover object-center']}
        />
      </div>
      <div class:list={['flex flex-col justify-center']}>
        <h1 class:list={['mb-2 text-base xs:text-3xl sm:text-4xl']}>{site.data.title}</h1>
        <p class:list={['hidden text-base sm:block']}>
          {site.data.tagline}
        </p>
      </div>
    </a>
  </div>
</header>
