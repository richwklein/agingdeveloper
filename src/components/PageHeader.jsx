import React from "react";
import {AppBar, Avatar, Box, ButtonBase, Toolbar, Typography} from "@mui/material";
import {GatsbyImage} from "gatsby-plugin-image";
import PropTypes from "prop-types";
import InnerContainer from "./InnerContainer";
import InternalLink from "./InternalLink";

/**
 * React component used to render the page avatar.
 *
 * @param {PageAvatarProps} props - The page avatar props.
 * @return {React.ReactElement} - The react component
 *
 * @example
 * <PageAvatar image={image} title={title} />
 */
export const PageAvatar = ({image, title}) => {
  return (<Avatar
    variant="rounded"
    sx={(theme) => ({
      bgcolor: "primary.dark",
      borderWidth: 1,
      borderColor: "secondary.light",
      borderStyle: "solid",
      [theme.breakpoints.up("sm")]: {
        width: "72px",
        height: "72px",
      },
      [theme.breakpoints.down("sm")]: {
        width: "48px",
        height: "48px",
      },
    })}>
    <GatsbyImage image={image} sx={{backgroundColor: "inherit"}} alt={title} />
  </Avatar>);
};

/**
 * @typedef PageAvatarProps - The page avatar props.
 * @property {GatsbyImageData} image - The gatsby image data for the site avatar.
 * @property {string} title - The title of the site.
 */
PageAvatar.propTypes = {
  image: PropTypes.object.isRequired,
  title: PropTypes.string.isRequired,
};

/**
 * React component used to render the page header.
 *
 * @param {PageHeaderProps} props - The page header props.
 * @return {React.ReactElement} - The react component
 *
 * @example
 * <PageHeader
 *   title={title}
 *   tagline={tagline}
 *   avatar={avatar}
 * />
 */
export const PageHeader = ({title, tagline, avatar}) => {
  return (
    <AppBar
      position="fixed"
      component="header"
      elevation={0}
      sx={{
        bgcolor: "primary.dark",
        color: "primary.contrastText",
        p: 2,
      }}
    >
      <InnerContainer>
        <Toolbar disableGutters>
          <ButtonBase component={InternalLink} to={"/"}
            sx={{color: "inherit", textDecoration: "none"}}>
            <PageAvatar image={avatar} title={title} />
            <Box sx={{ml: 2}}>
              <Typography variant="h4" noWrap={true} sx={{mb: 0.5}}>
                {title}
              </Typography>
              <Typography variant="h6" sx={(theme) => ({
                fontSize: "1rem",
                lineHeight: 1.4,
                [theme.breakpoints.up("sm")]: {
                  display: "block",
                },
                [theme.breakpoints.down("sm")]: {
                  display: "none",
                },
              })}>
                {tagline}
              </Typography>
            </Box>
          </ButtonBase>
        </Toolbar>
      </InnerContainer>
    </AppBar>
  );
};

/**
 * @typedef PageHeaderProps - The page header props.
 * @property {string} title - The title of the site.
 * @property {string} tagline - The tagline for the site.
 * @property {GatsbyImageData} avatar - The gatsby image data for the site avatar.
 */
PageHeader.propTypes = {
  title: PropTypes.string.isRequired,
  tagline: PropTypes.string.isRequired,
  avatar: PropTypes.object.isRequired,
};

export default PageHeader;
