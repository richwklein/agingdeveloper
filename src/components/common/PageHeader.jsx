import React from "react";
import {AppBar, Box, Typography} from "@mui/material";
import PropTypes from "prop-types";

import InnerContainer from "./InnerContainer";

const PageHeader = ({title, tagline}) => {
  return (
    <AppBar
      position="sticky"
      component="header"
      sx={{
        color: "primary.contrastText",
        backgroundColor: "primary.dark",
        padding: 2,
        marginBottom: 4,
      }}
    >
      <InnerContainer display="flex" alignItems="center">
        <Box width="100%">
          <Typography variant="h4" sx={{marginBottom: 0.5}}>
            {title}
          </Typography>
          <Typography variant="h6" sx={{fontSize: "1rem", lineHeight: 1.4}}>
            {tagline}
          </Typography>
        </Box>
      </InnerContainer>
    </AppBar>
  );
};

PageHeader.propTypes = {
  title: PropTypes.string.isRequired,
  tagline: PropTypes.string.isRequired,
};

export default PageHeader;
