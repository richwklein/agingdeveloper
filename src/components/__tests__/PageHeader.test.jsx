import React from "react";
import {cleanup, render} from "@testing-library/react";
import PageHeader from "../PageHeader";

// TODO improve test
describe("PageHeader", () => {
  afterEach(cleanup);

  const renderComponent = (href) => render(
      <PageHeader title="Test title" tagline="Test tagline is the best tagline"/>,
  );

  test("that it renders correctly.", () => {
    const {container} = renderComponent();
    expect(container.firstChild).toMatchSnapshot();
  });
});
