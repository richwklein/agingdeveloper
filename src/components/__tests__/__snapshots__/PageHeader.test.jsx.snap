// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader that it renders correctly. 1`] = `
<header
  class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionSticky css-nrystc-MuiPaper-root-MuiAppBar-root"
>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways MuiButtonBase-root css-1hyky9h-MuiTypography-root-MuiLink-root-MuiButtonBase-root"
        data-link="internal"
        href="/"
        tabindex="0"
      >
        <div
          class="MuiAvatar-root MuiAvatar-rounded MuiAvatar-colorDefault css-1qmj1zq-MuiAvatar-root"
        />
        <div
          class="MuiBox-root css-d0uhtl"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-725ddr-MuiTypography-root"
          >
            Test title
          </h4>
          <h6
            class="MuiTypography-root MuiTypography-h6 css-1ry1zvy-MuiTypography-root"
          >
            Test tagline is the best tagline
          </h6>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</header>
`;
